python/lsst/ts/ess/csc/data_client/young_32400_weather_station_data_client.py

```diff
@@ -629,57 +625,35 @@ def restart_rain_stopped_timer(self) -> None:
 
-    async def read_loop(self) -> None:
+    async def read_data(self) -> None:
         """Read raw data from the weather station.
 
         The format is as described by by DATA_REGEX.
         """
-        # Start the "rain stopped" timer so we can report "no rain"
-        # as soon after starting as it is safe to do so.
-        self.restart_rain_stopped_timer()
+        assert self.client is not None  # make mypy happy
+        read_bytes = await asyncio.wait_for(
+            self.client.readuntil(YOUNG_TERMINATOR),
+            timeout=self.config.read_timeout,
+        )
+        self.num_consecutive_read_timeouts = 0
+        data = read_bytes.decode().strip()
+        if not data:
+            return
+        match = DATA_REGEX.fullmatch(data)
+        if match is None:
+            self.log.warning(f"Ignoring {data=}: could not parse the data")
+            return
+        # Convert raw data values from str to int.
+        raw_data_dict = {key: int(value) for key, value in match.groupdict().items()}
         try:
-            while self.connected:
-                assert self.client is not None  # make mypy happy
-                read_bytes = await asyncio.wait_for(
-                    self.client.readuntil(YOUNG_TERMINATOR),
-                    timeout=self.config.read_timeout,
-                )
-                self.num_consecutive_read_timeouts = 0
-                data = read_bytes.decode().strip()
-                if not data:
-                    continue
-                match = DATA_REGEX.fullmatch(data)
-                if match is None:
-                    self.log.warning(f"Ignoring {data=}: could not parse the data")
-                    continue
-                # Convert raw data values from str to int.
-                raw_data_dict = {
-                    key: int(value) for key, value in match.groupdict().items()
-                }
-                try:
-                    await self.handle_data(**raw_data_dict)
-                except Exception as e:
-                    self.log.exception(f"Failed to handle {data=}: {e!r}")
-        except asyncio.TimeoutError:
-            self.num_consecutive_read_timeouts += 1
-            self.log.warning(
-                f"Read timed out. This is timeout #{self.num_consecutive_read_timeouts} "
-                f"of {self.config.max_read_timeouts} allowed."
-            )
-            if self.num_consecutive_read_timeouts >= self.config.max_read_timeouts:
-                self.log.error(
-                    f"Encountered at least {self.config.max_read_timeouts} timeouts. Raising error."
-                )
-                raise
-        except StopIteration:
-            self.log.info("read loop ends: out of simulated raw data")
+            await self.handle_data(**raw_data_dict)
         except Exception as e:
-            self.log.exception(f"read loop failed: {e!r}")
-            raise
+            self.log.exception(f"Failed to handle {data=}: {e!r}")
```

Comment:
Failure to handle data that matches a regex is pretty disturbing, so I hope we will never see this. But if it does occur I think this should raise (which will send the CSC to fault?).

If the base class logs an exception then I suggest omitting the try/except.

Reply:
The base class logs the exception so I removed the try/except.